app/controllers/application_controller.rb

```diff
@@ -35,6 +37,15 @@ def current_user_info
     current_user['info']
   end
 
+  def set_has_multiple_providers
+    @has_multiple_providers = has_multiple_providers?
+  end
+
+  def has_multiple_providers?
+    provider_count = session.try(:[], :auth_user).try(:[], "provider_count")
```

Comment:
You can use `#dig` instead of `try`, it's a bit nicer perhaps? Also `fetch` if you want a default if it's `nil`, to avoid to `||` below?

Reply:
I tried `dig` and it didn't work correctly. I'll try again.

Reply:
Now `dig`ging, figured it out, cheers :)